deeppavlov/models/sklearn/sklearn_component.py

```diff
@@ -0,0 +1,315 @@
+                else:
+                    log.info("Converting input for model {} to sparse array".format(self.model_class))
+                    self.model.fit(csr_matrix(x_features), y_)
+            except:
+                raise ConfigError("Can not fit on the given data".format(self.model_class))
+
+        return
+
+    def __call__(self, *args):
+        """
+        Infer on the given data according to given in the config infer method, \
+            e.g. ``"predict", "predict_proba", "transform"``
+
+        Args:
+            *args: list of inputs
+
+        Returns:
+            predictions, e.g. list of labels, array of probability distribution, sparse array of vectorized samples
+        """
+        x_features = self.compose_input_data(args)
+
+        try:
+            predictions = getattr(self.model, self.infer_method)(x_features)
+        except:
+            try:
+                if issparse(x_features):
+                    log.info("Converting input for model {} to dense array".format(self.model_class))
+                    predictions = getattr(self.model, self.infer_method)(x_features.todense())
+                else:
+                    log.info("Converting input for model {} to sparse array".format(self.model_class))
+                    predictions = getattr(self.model, self.infer_method)(csr_matrix(x_features))
+            except:
+                raise ConfigError("Can not infer on the given data".format(self.model_class))
+
+        if isinstance(predictions, list):
+            #  ``predict_proba`` sometimes returns list of n_outputs (each output corresponds to a label)
+            #  but we will return (n_samples, n_labels)
+            #  where each value is a probability of a sample to belong with the label
+            predictions_ = [[predictions[j][i][1] for j in range(len(predictions))] for i in range(x_features.shape[0])]
+            predictions = np.array(predictions_)
+
+        if len(predictions.shape) == 1:
+            predictions = predictions.reshape(-1, 1)
+        return predictions
+
+    def init_from_scratch(self) -> None:
+        """
+        Initialize ``self.model`` as some sklearn model from scratch with given in ``self.model_params`` parameters.
+
+        Returns:
+            None
+        """
+        log.info("Initializing model {} from scratch".format(self.model_class))
+        model_function = cls_from_str(self.model_class)
+
+        if model_function is None:
+            raise ConfigError("Model with {} model_class was not found.".format(self.model_class))
+
+        given_params = {}
+        if self.model_params:
+            available_params = self.get_function_params(model_function)
+            for param_name in self.model_params.keys():
+                if param_name in available_params:
+                    try:
+                        given_params[param_name] = cls_from_str(self.model_params[param_name])
+                    except:
```

Comment:
What kinds of exceptions are you catching here?

Reply:
AttributeError

Reply:
`except AttributeError:`?